quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HttpPropertyMappers.java

```diff
@@ -223,24 +215,18 @@ private static File getDefaultKeystorePathValue() {
         return null;
     }
 
-    private static Optional<String> resolveKeyStoreType(Optional<String> value,
+    private static String resolveKeyStoreType(String value,
             ConfigSourceInterceptorContext configSourceInterceptorContext) {
-        if (value.isPresent()) {
-            try {
-                if (FipsMode.valueOfOption(value.get()).equals(FipsMode.STRICT)) {
-                    return of("BCFKS");
-                }
-                return empty();
-            } catch (IllegalArgumentException ignore) {
-            }
+        if (FipsMode.valueOfOption(value).equals(FipsMode.STRICT)) {
+            return "BCFKS";
```

Comment:
```suggestion
       try {
            if (FipsMode.valueOfOption(value).equals(FipsMode.STRICT)) {
                return "BCFKS";
            }
        } catch (IllegalArgumentException ignore) {
```

I think the Enum's IllegalArgumentEx should stay in this case.

Reply:
The option validation will happen before this, so we're assured that it's a valid FipsMode. To remove any possible exceptions, I've simplified both this and the fips-mode logic.

Reply:
Got it. It's better now - more defensive. Thanks